src/Common/SettingsChanges.cpp

```diff
@@ -46,4 +46,29 @@ Field * SettingsChanges::tryGet(std::string_view name)
     return &change->value;
 }
 
+bool SettingsChanges::insertSetting(std::string_view name, const Field & value)
+{
+    if (std::find_if(begin(), end(), [&name](const SettingChange & change) { return change.name == name; }) != end())
+        return false;
+    emplace_back(name, value);
+    return true;
+}
+
+void SettingsChanges::setSetting(std::string_view name, const Field & value)
+{
+    if (auto * v = tryGet(name))
```

Comment:
Consider rename `v` variable into something more informative. For example `setting_value`.